### ISSUE TITLE
1.4.2.1 (and 1.4.2 for that matter) aren't pulling in core-js dependencies via babel-runtime

### ISSUE THREAD

**Issue by andylash** [meteor/meteor#8042]
This looks exactly like issue 7833.  I tried following the details there and see a similar thing:

⚡ cat 1.4.1/.meteor/local/build/programs/server/packages/modules.js | wc
  148051  525484 5122880
⚡ cat 1.4.2/.meteor/local/build/programs/server/packages/modules.js | wc
   22643   91575  782755

Have tried complete rebuild from source, so not a node_modules or reset fixable thing.  I presume this is related somehow to babel-runtime change, but haven't been able to sort it out.

 Error: Cannot find module '../core-js/symbol/iterator'
W20161110-11:08:15.524(-8)? (STDERR)     at require (packages/modules-runtime.js:109:19)
W20161110-11:08:15.524(-8)? (STDERR)     at meteorInstall.node_modules.meteor.opstarts-calcengine.node_modules.babel-runtime.helpers.typeof.js (/Users/andy/meteorsrc/opstarts3/app/.meteor/local/build/programs/server/packages/opstarts-calcengine.js:39493:17)
W20161110-11:08:15.525(-8)? (STDERR)     at fileEvaluate (packages/modules-runtime.js:181:9)
W20161110-11:08:15.526(-8)? (STDERR)     at Module.require (packages/modules-runtime.js:106:16)
W20161110-11:08:15.526(-8)? (STDERR)     at Module.Mp.import (/Users/andy/.meteor/packages/modules/.0.7.7.fju5jr++os+web.browser+web.cordova/npm/node_modules/reify/lib/runtime.js:70:16)
W20161110-11:08:15.527(-8)? (STDERR)     at meteorInstall.node_modules.meteor.opstarts-calcengine.js.common.money_conversions.js (packages/opstarts-calcengine/js/common/money_conversions.js:2:1)
W20161110-11:08:15.527(-8)? (STDERR)     at fileEvaluate (packages/modules-runtime.js:181:9)
W20161110-11:08:15.528(-8)? (STDERR)     at Module.require (packages/modules-runtime.js:106:16)
W20161110-11:08:15.529(-8)? (STDERR)     at Module.Mp.import (/Users/andy/.meteor/packages/modules/.0.7.7.fju5jr++os+web.browser+web.cordova/npm/node_modules/reify/lib/runtime.js:70:16)
W20161110-11:08:15.529(-8)? (STDERR)     at meteorInstall.node_modules.meteor.opstarts-calcengine.js.server.index.js (packages/opstarts-calcengine/js/server/index.js:1:1)
W20161110-11:08:15.529(-8)? (STDERR)     at fileEvaluate (packages/modules-runtime.js:181:9)
W20161110-11:08:15.530(-8)? (STDERR)     at require (packages/modules-runtime.js:106:16)
W20161110-11:08:15.530(-8)? (STDERR)     at /Users/andy/meteorsrc/opstarts3/app/.meteor/local/build/programs/server/packages/opstarts-calcengine.js:49676:15
W20161110-11:08:15.530(-8)? (STDERR)     at /Users/andy/meteorsrc/opstarts3/app/.meteor/local/build/programs/server/packages/opstarts-calcengine.js:49682:3
W20161110-11:08:15.530(-8)? (STDERR)     at /Users/andy/meteorsrc/opstarts3/app/.meteor/local/build/programs/server/boot.js:295:34
W20161110-11:08:15.531(-8)? (STDERR)     at Array.forEach (native)

This is on MacOs, version: 10.11.6.

"dependencies": {
    "@opstarts/basic-number-format": "0.1.2",
    "ag-grid": "5.4.x",
    "ag-grid-enterprise": "5.4.x",
    "ag-grid-react": "5.4.x",
    "autobind-decorator": "^1.3.3",
    "babel-plugin-transform-class-properties": "^6.9.1",
    "babel-plugin-transform-decorators-legacy": "^1.3.4",
    "babel-plugin-transform-export-extensions": "^6.8.0",
    "babel-preset-meteor": "^6.13.0",
    "babel-preset-react": "^6.5.0",
    "babel-runtime": "^6.18.0",
    "bcrypt": "^0.8.7",
    "bcrypt-pbkdf": "^1.0.0",
    "bowser": "^1.4.6",
    "classnames": "~2.2.3",
    "comma-separated-values": "3.6.4",
    "d3": "3.5.16",
    "dagre-d3": "0.4.17",
    "deep-rename-keys": "0.2.0",
    "dnd-core": "^2.0.2",
    "ecc-jsbn": "^0.1.1",
    "externalify": "0.1.0",
    "graphlib": "2.1.1",
    "hopscotch": "0.2.5",
    "invariant": "2.2.1",
    "jodid25519": "^1.0.2",
    "jquery": "1.11.2",
    "jsbn": "^0.1.0",
    "jszip": "2.4.0",
    "lodash": "3.10.1",
    "material-ui": "0.16.0",
    "memory-cache": "0.1.6",
    "meteor-node-stubs": "0.2.3",
    "mobx": "2.6.0",
    "mobx-react": "3.5.8",
    "moment": "2.15.1",
    "money": "0.2.0",
    "node-quickbooks": "2.0.8",
    "papaparse": "^4.1.2",
    "parse-uri": "1.0.0",
    "promise-queue": "^2.2.3",
    "react": "15.3.2",
    "react-addons-css-transition-group": "15.3.2",
    "react-addons-perf": "^15.2.0",
    "react-addons-pure-render-mixin": "15.3.2",
    "react-addons-transition-group": "15.3.2",
    "react-click-outside": "2.1.0",
    "react-dnd": "2.1.4",
    "react-dnd-html5-backend": "2.1.2",
    "react-dom": "15.3.2",
    "react-dropzone": "^3.5.3",
    "react-select": "^1.0.0-rc.2",
    "react-tap-event-plugin": "^1.0.0",
    "stripe": "2.8.0",
    "xlsx-browserify-shim": "0.0.5"
  },
  "devDependencies": {
    "chimp": "^0.41.1",
    "enzyme": "^2.4.1",
    "mobx-react-devtools": "^4.2.6",
    "redbox-react": "1.2.2"
  }

.meteor/versions: 
3stack:match-library@1.0.5
accounts-admin-ui-bootstrap-3@0.2.9
accounts-base@1.2.14
accounts-oauth@1.1.15
accounts-password@1.3.2
alanning:roles@1.2.15
aldeed:collection2@2.9.1
aldeed:collection2-core@1.1.1
aldeed:schema-deny@1.0.1
aldeed:schema-index@1.0.1
aldeed:simple-schema@1.5.3
allow-deny@1.0.5
almogdesign:parsley.js@1.1.7
andylash:accounts-quickbooks@0.1.3
andylash:basic-logger@2.0.3
andylash:basic-tags@0.1.1
andylash:quickbooks@0.1.1
arillo:flow-router-helpers@0.5.2
audit-argument-checks@1.0.7
autoupdate@1.3.12
babel-compiler@6.13.0
babel-runtime@1.0.0
base64@1.0.10
billyvg:accounts-stripe@0.1.7
billyvg:stripe-oauth@0.1.7
binary-heap@1.0.10
blaze@2.2.0
blaze-html-templates@1.0.5
blaze-tools@1.0.10
boilerplate-generator@1.0.11
bozhao:link-accounts@1.2.9
browser-policy@1.0.9
browser-policy-common@1.0.11
browser-policy-content@1.0.12
browser-policy-framing@1.0.12
caching-compiler@1.1.9
caching-html-compiler@1.0.7
callback-hook@1.0.10
check@1.2.4
coffeescript@1.11.1_3
cosmos:browserify@0.8.4
d3js:d3@3.5.8
ddp@1.2.5
ddp-client@1.3.2
ddp-common@1.2.8
ddp-rate-limiter@1.0.6
ddp-server@1.3.12
deanius:promise@3.1.3
deps@1.0.12
dfischer:phantomjs@1.0.4
diff-sequence@1.0.7
dispatch:mocha-browser@0.0.4
dispatch:mocha-phantomjs@0.1.7
dispatch:phantomjs-tests@0.0.5
ecmascript@0.6.0
ecmascript-collections@0.1.6
ecmascript-runtime@0.3.15
ejson@1.0.13
email@1.1.18
erasaur:meteor-lodash@4.0.0
fastclick@1.0.13
force-ssl@1.0.13
fortawesome:fontawesome@4.7.0
frozeman:q@1.1.1
geojson-utils@1.0.10
gfk:notifications@1.1.4
google@1.0.1
google-apps-script@1.0.0
hot-code-push@1.0.4
html-tools@1.0.11
htmljs@1.0.11
http@1.2.10
id-map@1.0.9
jamiter:pluralize@1.0.2
jquery@1.11.10
kadira:blaze-layout@2.3.0
kadira:debug@3.2.2
kadira:flow-router@2.12.1
kadira:runtime-dev@0.0.1
launch-screen@1.0.12
less@2.5.7
livedata@1.0.18
localstorage@1.0.12
logging@1.1.16
matb33:collection-hooks@0.8.4
mdg:validated-method@1.1.0
mdg:validation-error@0.2.0
meteor@1.6.0
meteor-base@1.0.4
meteor-platform@1.2.6
meteorhacks:inject-initial@1.0.4
meteorhacks:kadira@2.30.2
meteorhacks:kadira-binary-deps@1.5.0
meteorhacks:kadira-profiler@1.3.0
meteorhacks:meteorx@1.4.1
meteorhacks:picker@1.0.3
meteorhacks:subs-manager@1.6.4
meteorhacks:unblock@1.1.0
mikowals:batch-insert@1.1.14
minifier-css@1.2.15
minifier-js@1.2.15
minimongo@1.0.18
mizzao:timesync@0.4.0
mizzao:user-status@0.6.6
mobile-status-bar@1.0.13
modules@0.7.7
modules-runtime@0.7.7
momentjs:moment@2.8.4
mongo@1.1.14
mongo-id@1.0.6
mongo-livedata@1.0.12
mrt:later@1.6.1
nitrolabs:cdn@1.2.6
npm-bcrypt@0.9.2
npm-mongo@2.2.11_2
oauth@1.1.12
oauth-encryption@1.2.1
oauth1@1.1.11
oauth2@1.1.11
observe-sequence@1.0.14
opstarts-calcengine@2.0.0
opstarts-constants@1.0.0
opstarts-importer@0.0.0
opstarts-intelligence@0.0.0
opstarts:fixtures@0.0.1
ordered-dict@1.0.9
pascoual:otp@0.1.3
pcel:loading@1.0.3
peerlibrary:assert@0.2.5
peerlibrary:blocking@0.5.2
peerlibrary:directcollection@0.6.2
peerlibrary:xml2js@0.4.8_1
peppelg:bootstrap-3-modal@1.0.3
percolate:momentum@0.7.2
percolate:synced-cron@1.3.2
percolate:velocityjs@1.1.0
percolatestudio:percolatestudio-migrations@0.7.2
percolatestudio:segment.io@2.0.0_1
pfafman:filesaver@0.2.3_2
practicalmeteor:mocha-core@1.0.1
promise@0.8.8
raix:eventemitter@0.1.3
random@1.0.10
rate-limit@1.0.6
react-meteor-data@0.2.9
react-template-helper@0.2.9
reactive-dict@1.1.8
reactive-var@1.0.11
reload@1.1.11
retry@1.0.9
routepolicy@1.0.12
sergeyt:typeahead@0.11.1_9
service-configuration@1.0.11
session@1.1.7
sha@1.0.9
shell-server@0.2.1
spacebars@1.0.13
spacebars-compiler@1.0.13
srp@1.0.10
standard-app-packages@1.0.9
standard-minifier-css@1.3.2
standard-minifier-js@1.2.1
stylus@2.513.7
tarang:zip@0.1.0
templating@1.2.15
templating-compiler@1.2.15
templating-runtime@1.2.15
templating-tools@1.0.5
thepumpinglemma:flat@1.1.0
tmeasday:check-npm-versions@0.3.1
tmeasday:publish-counts@0.8.0
tracker@1.1.1
twbs:bootstrap@3.3.6
ui@1.0.12
underscore@1.0.10
url@1.0.11
url2@1.0.2
vsivsi:job-collection@1.4.0
webapp@1.3.12
webapp-hashing@1.0.9
xolvio:backdoor@0.2.1
zimme:active-route@2.3.2
zimme:underscore-contrib@0.3.0

**Comment by benjamn** [comment]
Thanks for the information you've provided, but nothing jumps out at me that would explain the problem. Can you provide any sort of reproduction?

**Comment by bluebackground** [comment]
I'm new to web development and meteor, but I wanted to add a note that I've been running into what looks like a similar issue. After creating a meteor project using "meteor create newapp" and try running it, I get an error message that has a lot of text that looks similar to the first post. One line is: "Error: The babel-runtime npm package could not be found in your node_modules directory. Please run the following command to install it: meteor npm install --save babel-runtime". This fixes the issue. I don't know how or why this started though. I used to run meteor without many problems prior to 1.4.2.1 update.

**Comment by benjamn** [comment]
@bluebackground The warning you're seeing was added in Meteor 1.4.2.1, and it's an intentional change. Any app that uses the `ecmascript` or `coffeescript` packages now needs to depend on the `babel-runtime` npm package.

Even before 1.4.2.1, whenever you created a new app, you needed to run `meteor npm install`. With Meteor 1.4.2.1, the `babel-runtime` package is listed as a dependency in the `package.json` file for new apps, along with some other basic dependencies, so `meteor npm install` will install it for you.

**Comment by Cypher1** [comment]
I'm added the babel-runtime npm package (and installed it) via npm install babel-runtime --save-dev -g (global as I will be using it in other projects), but I'm still getting this error. Is this a bug in one of my dependencies?

How do we fix it? Thanks (Little bit annoyed that this behaviour was changed without adding error handling for it)

**Comment by Maxpain177** [comment]
@Cypher1
Try meteor
```npm install babel-runtime --save``` (not ```--save-dev```)

**Comment by Cypher1** [comment]
Doesn't seem to have made a difference but thanks for trying

**Comment by Cypher1** [comment]
According to another issue (https://github.com/meteor/meteor/issues/6918) manual reinstallation is the simple solution (once bable is installed). I guess I'll try that.

**Comment by ptica** [comment]
`Created a new Meteor app in 'directory'.

To run your new app:
  cd directory
  meteor`

this message should be probably updated as well to include the `meteor npm install` step,
as it may confuse the newcomers

**Comment by Cypher1** [comment]
Tried the manual installation process, I'm guessing that it's in a dependency for my project. For now I'm just reverting to the previous meteor.

**Comment by Cypher1** [comment]
Thanks @Maxpain177 I think the meteor npm install is different to npm install somehow. Thanks for the help.

**Comment by benjamn** [comment]
@Cypher1 The biggest difference is that `meteor npm ...` always uses the same version of Node that Meteor uses (see `meteor node --version`), whereas the version of `npm` that you have installed globally on your system could be using any version of Node.

**Comment by andylash** [comment]
OK, I finally sorted my original problem out.  It took a while.  Basically I had a private meteor package, calcengine, with a Npm.depends call in its package.js.  I removed that entirely and replaced the npm packages referenced there with references to them in calcengine's package.json file.  Not sure why that appears to have addressed this.

**Comment by bluebackground** [comment]
@benjamn Thanks for telling me about hte the new step "meteor npm install". The only comment I have on that is that there are many meteor tutorials done all across the web that show just "meteor create new app, cd newapp, meteor". Even though it's an easy step, it seemed to be what meteor was known for - easy to get up and running with a basic setup (add as needed).

**Comment by hwillson** [comment]
Hi @bluebackground - there is an open issue to track this feature request; see #6848. Please add your support there (as this current issue thread has been closed). Thanks!
